### DISCUSSION TITLE
[Projeto Angular] Style Guide

### DISCUSSION THREAD

**Discussion by magnolemos** [frontendbr/forum#1343] [Perguntas e Respostas]
Boa tarde Galera!

Comecei a usar o Angular no final do ano passado e estou muito satisfeito com os resultados. Porém, sinto falta de um style guide para manter o projeto mais inteligível e padronizado (apesar do angular já ser bastante organizado).

No React usava o do Airbnb. Em razão do typeScript, ele não faz sentido para projeto Angular.

Sendo assim, gostaria de ouvir a experiencia de vocês. 
Utilizam algum? Não vê necessidade? Apenas o guia do Angular?

obs: obrigado a todos que colaboraram no tópico do lazy loading. Segui a maioria das dicas e o projeto tá ficando bem massa 💃

**Comment by thaymara** [comment]
Oi magnolemos!
No projeto que estou trabalhando atualmente nós usamos o guia do Angular mesmo e está funcionando até que bem. :)

**Comment by magnolemos** [comment]
@thaymara , valeu demais por compartilhar!

Com React eu conseguia configurar para da warning no console se não estivesse no padrão do AirBnb. Senti falta disso no angular.

Acredito que irei fazer como vocês mesmo.   Valeu demais!

**Comment by menosprezzi** [comment]
Pessoal, vocês estão se referindo à um padrão de projeto ou de qualidade de
código?
Padrão de projeto sempre segui a modularização sugerida pelo próprio
Angular. E sobre qualidade em angular, o pessoal da comunidade já adotou o
plugin https://github.com/mgechev/codelyzer para o lint. O codelyzer é um
conjunto de regras de lint, como o do Airbnb para o react. Todos os
projetos Angular que começam a partir do starter deles já vem com esse
plugin no package.json. Basta vocês aumentarem as regras desse lint na
configuração do tslint dos seus projetos. Funciona bem.

**Comment by williangd** [comment]
Uso também o prettier (https://prettier.io/), pra formatar os arquivos ao salvar. É bom para manter o padrão em todos os arquivos.

Caso queira alterar o padrão usado por ele, é bem simples também.

Único problema que vejo, que as vezes ele é atualizado e os padrões mudam um pouco haha.

**Comment by magnolemos** [comment]
@menosprezzi , qualidade de código mesmo. Era exatamente o codelyzer  que eu procurava, muito obrigado!

@williangd, mão na roda hein? Também não conhecia. Vou testar ele aqui. Valeu demais!

**Comment by magnolemos** [comment]
Galera, vim aqui só agradecer. Utilizei o codelyzer junto com o prettier. Era o que eu precisa. Valeu mesmo :D : D : D

**Comment by menosprezzi** [comment]
Showw @williangd, sempre ouvi falar no prettier mas não sabia que ele era tão bom na parte de compatibilidade entre editores!
